inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -742,15 +777,51 @@ FunctionsComparator::Result FunctionsComparator::compare(
                 auto const1 = ngraph::as_type_ptr<Constant>(node1->get_input_node_shared_ptr(i));
                 auto const2 = ngraph::as_type_ptr<Constant>(node2->get_input_node_shared_ptr(i));
 
-                const auto equal = [](std::shared_ptr<Constant> c1, std::shared_ptr<Constant> c2) {
-                    const auto& c1v = c1->cast_vector<double>();
-                    const auto& c2v = c2->cast_vector<double>();
-
-                    return c1v.size() == c2v.size() && std::equal(
-                                                           begin(c1v), end(c1v), begin(c2v),
-                                                           [](const double& s1, const double& s2) {
-                                                               return std::abs(s1 - s2) < 0.001;
-                                                           });
+                const auto equal = [](std::shared_ptr<Constant> c1,
+                                      std::shared_ptr<Constant> c2) {
+                    using namespace ::attr_comparison::equal;
+                    const auto c1t = c1->get_element_type();
+                    const auto c2t = c2->get_element_type();
+                    if (c1t != c2t) {
+                        return false;
+                    }
+
+                    bool rc = false;
+                    switch (c1t) {
+                    case ngraph::element::Type_t::bf16: {
+                        auto c1v = c1->cast_vector<ngraph::bfloat16>();
+                        auto c2v = c2->cast_vector<ngraph::bfloat16>();
+                        rc = c1v.size() == c2v.size()
+                            ? Equal<std::vector<ngraph::bfloat16>>::equal_value(c1v, c2v)
+                            : false;
+                        break;
+                    }
+                    case ngraph::element::Type_t::f16: {
+                        const auto &c1v = c1->cast_vector<ngraph::float16>();
+                        const auto &c2v = c2->cast_vector<ngraph::float16>();
+                        rc = c1v.size() == c2v.size()
```

Comment:
Don't see reason to use elvis operator.
This should works:
```
return Equal<std::vector<float16>>::equal_value(c1v, c2v);
```
1) `Equal` for `vector` check size
2) `rc` is not need
3) `c1v.size() == c2v.size() || Equal<vec>(c1v, c2v)` looks better.